src/pudl/output/eia923.py

```diff
@@ -475,7 +475,13 @@ def boiler_fuel_eia923(pudl_engine, freq=None, start_date=None, end_date=None):
     )
 
     # Create a date index for grouping based on freq
-    by = ["plant_id_eia", "boiler_id", "energy_source_code", "fuel_type_code_pudl"]
+    by = [
+        "plant_id_eia",
+        "boiler_id",
+        "energy_source_code",
+        "fuel_type_code_pudl",
+        "prime_mover_code",
+    ]
```

Comment:
needed to add the pm code in this output table `by` because we use this in a groupby when `freq != None`